File: my-folder/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int lp = 0;
        int rp = height.length - 1;
        int maxArea = 0;

        while(lp < rp){
            int h = height[lp] < height[rp] ? height[lp] : height[rp];
            int l = rp - lp;
            int currentArea = l*h;
            maxArea = currentArea > maxArea ? currentArea : maxArea;

            // if(height[lp] > height[rp]){
            //     rp --;
            // }else{
            //     lp ++;
            // }

            while(height[lp] <= h && lp < rp) lp ++;
            while(height[rp] <= h && lp < rp) rp --;
        }

        return maxArea;
    }
}

File: my-folder/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = -inf
        start =  0
        end = len(height) - 1
        
        while start < end:
            area = min(height[start], height[end]) * (end - start)
            maxArea = max(area, maxArea)
                   
            if height[start] < height[end]:
                start += 1
            else:
                end -= 1

        return maxArea

